benchmark/test/test_framework.py.in

```diff
@@ -0,0 +1,250 @@
+#!/usr/bin/env python3
+import subprocess
+import difflib
+import json
+import typing
+import re
+import pathlib
+import sys
+
+sourcepath = pathlib.Path("@CMAKE_CURRENT_SOURCE_DIR@")
+binpath = pathlib.Path("@PROJECT_BINARY_DIR@")
```

Comment:
Together with MPI_EXECUTABLE and MPI_NUMPROC_FLAG, this would be 4 variables that need to be passed as arguments - which approach would you consider cleaner?